euclidlib/photo/_le3_2pcf_wl.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from os import PathLike
+from typing import TYPE_CHECKING, Tuple, Union
+from numpy.typing import NDArray
+
+from euclidlib.photo import photo_data
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+    from typing import Any, TypeAlias
+
+
+# type alias
+_DictKey: "TypeAlias" = Union[str, int, Tuple["_DictKey", ...]]
+
+def xi_tpcf(
+    path: str | PathLike[str],
+    *,
+    include: Sequence[tuple[Any, ...]] | None = None,
+    exclude: Sequence[tuple[Any, ...]] | None = None,
+) -> dict[_DictKey, NDArray[Any]]:
+    """
+    Read the Weak Lensing two-point correlation functions for LE3.
+    """
+
+    corr_str = "ShearShear"
+
+    # Load LE3 output file: 2PCF
+    tpcf_le3 = photo_data.TpcfDataReader.from_fits(
+        path,
+        corr_str=corr_str,
+    )
+
+    z_combinations = list(tpcf_le3._data[corr_str].keys())
+
+    # Get LE3 data
+    xis = {}
+    for component in ("+", "-"):
+        xis[component] = {}
+        for i, z_comb in enumerate(z_combinations):
+            xis[component][z_comb] = tpcf_le3.get_xi_component(component,
+                                                               z_comb)
```

Comment:
We are only loading the xi correlation functions here -- could we load w(θ) or γ(θ) at the same time? Or are they provided in different files?

Can we load and return `THETA`, `THMIN`, `THMAX`, weight for each angular bin?

Reply:
Hello, in the new commit, I added THMIN, THMAX on the highest level of the dictionary. Theta is added as an array also on the highest level. Then the weight(theta) is loaded for each redshift combination. I hope that is what you had in mind. The unit test is updated as well.
For the other correlation functions, I don't have the FITS files for them so I could not test that directly. But we could contact Martin Kilbinger or someone else for them.